### PR TITLE
bpo-37165: Convert _count_elements to the argument clinic

### PR DESCRIPTION
**Pull request by rhettinger**
https://bugs.python.org/issue37165

### CODE REVIEW

File: Modules/_collectionsmodule.c [hunk 1]
```diff
@@ -2509,17 +2514,12 @@ PyDoc_STRVAR(module_doc,
 - defaultdict:  dict subclass with a default value factory\n\
 ");
 
-static struct PyMethodDef module_functions[] = {
```
Comment (pablogsal) [inline Modules/_collectionsmodule.c]: The argument clinic exports a macro in the generated header that we need to include here in turn (it contains the previous invocation but with the generated code and docstring). It should look like:

```C
static struct PyMethodDef module_functions[] = {
    _COLLECTIONS__COUNT_ELEMENTS_METHODDEF
    {NULL,       NULL}          /* sentinel */
};
```

File: Modules/_collectionsmodule.c [hunk 2]
```diff
@@ -2509,17 +2514,12 @@ PyDoc_STRVAR(module_doc,
...
-    {"_count_elements", _count_elements,    METH_VARARGS,   _count_elements_doc},
-    {NULL,       NULL}          /* sentinel */
-};
-
 static struct PyModuleDef _collectionsmodule = {
     PyModuleDef_HEAD_INIT,
     "_collections",
     module_doc,
     -1,
-    module_functions,
```
Comment (pablogsal) [inline Modules/_collectionsmodule.c]: This needs to be kept in order for the function to be exported in the module.

**Review by pablogsal — COMMENTED** [review]
Thanks for working on this :) 

I left some comments regarding a couple of missing things.

File: Modules/_collectionsmodule.c (hunk 1 above)
Comment (rhettinger) [reply Modules/_collectionsmodule.c]: Will do.

File: Modules/_collectionsmodule.c (hunk 2 above)
Comment (rhettinger) [reply Modules/_collectionsmodule.c]: Right.

**Review by pablogsal — APPROVED** [review]
LGTM

**Comment by serhiy-storchaka** [comment]
Usually we do not add a NEWS entry when convert a module or separate functions to Argument Clinic, as such conversion does not change the behavior. In this particular case the converted function is private, so there is even less reasons to add a NEWS entry.

**Pull request merged**